src/assistants/openai/openai_api_version.rs

```diff
@@ -0,0 +1,175 @@
+                // Try to create the header value from the bearer token
+                if let Ok(bearer_header) = HeaderValue::from_str(&format!("Bearer {api_key}")) {
+                    headers.insert("Authorization", bearer_header);
+                } else {
+                    headers.insert(
+                        "Error",
+                        HeaderValue::from_static("Invalid Authorization Header"),
+                    );
+                };
+                headers.insert("OpenAI-Beta", HeaderValue::from_static("assistants=v2"));
+            }
+            OpenAIAssistantVersion::Azure => {
+                // Try to create the header value from the bearer token
+                if let Ok(api_key_header) = HeaderValue::from_str(api_key) {
+                    headers.insert("api-key", api_key_header);
+                } else {
+                    headers.insert(
+                        "Error",
+                        HeaderValue::from_static("Invalid Authorization Header"),
+                    );
+                };
+            }
+        };
+        headers
+    }
+
+    pub(crate) fn get_tools_payload(&self) -> Value {
+        match self {
+            OpenAIAssistantVersion::V1 => json!([{
+                "type": "retrieval"
+            }]),
+            OpenAIAssistantVersion::V2 | OpenAIAssistantVersion::Azure => json!([{
+                "type": "file_search"
+            }]),
+        }
+    }
+
+    pub(crate) fn add_message_attachments(
+        &self,
+        message_payload: &Value,
+        file_ids: &[String],
+    ) -> Value {
+        let mut message_payload = message_payload.clone();
+        match self {
+            OpenAIAssistantVersion::V1 => {
+                message_payload["file_ids"] = json!(file_ids);
+            }
+            OpenAIAssistantVersion::V2 | OpenAIAssistantVersion::Azure => {
+                let file_search_json = json!({
+                    "type": "file_search"
+                });
+                let attachments_vec: Vec<Value> = file_ids
+                    .iter()
+                    .map(|file_id| {
+                        json!({
+                            "file_id": file_id.to_string(),
+                            "tools": [file_search_json.clone()]
+                        })
+                    })
+                    .collect();
+                message_payload["attachments"] = json!(attachments_vec);
+            }
+        }
+        message_payload
+    }
+}
+
+impl FromStr for OpenAIAssistantVersion {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "v1" => Ok(OpenAIAssistantVersion::V1),
+            "v2" => Ok(OpenAIAssistantVersion::V2),
+            "azure" => Ok(OpenAIAssistantVersion::Azure),
+            _ => Err(anyhow!("Invalid version: {}", s)),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Eq, PartialEq)]
```

Comment:
This enums encodes all the possible resources/endpoints associated with Assistants API.

Reply:
meat!!